zokrates_core/src/zir/mod.rs

```diff
@@ -556,6 +570,65 @@ impl<'ast, T: fmt::Debug> fmt::Debug for ZirExpressionList<'ast, T> {
 
 // Common behaviour accross expressions
 
+pub trait Expr<'ast, T> {
```

Comment:
@dark64 this may be useful to unify the treatment of expressions in zir propagation